src/lavinmq/amqp/queue/message_store.cr

```diff
@@ -279,10 +285,16 @@ module LavinMQ
       end
 
       private def open_new_segment(next_msg_size = 0) : MFile
-        @wfile.unmap unless @wfile == @rfile
+        unless @wfile_id.zero?
+          append_msg_count(@wfile, @segment_msg_count[@wfile_id])
+          @wfile.read_only = true
```

Comment:
This is only so that we don't by mistake edit any data in these files, but it adds a lot of code `MFile`, so that could be optional if we still want that in this PR.